backend/src/exercise/exercise-wrapper.ts

```diff
@@ -243,12 +243,23 @@ export class ExerciseWrapper extends NormalType<
         public readonly temporaryActionHistory: ActionWrapper[],
         databaseService: DatabaseService,
         private readonly stateVersion: number,
-        private readonly initialState = ExerciseState.create(),
+        private initialState = ExerciseState.create(),
         private currentState: ExerciseState = initialState
     ) {
         super(databaseService);
     }
 
+    public migrateStates(
+        update: (
+            initialState: unknown,
+            currentState: unknown
+        ) => { initialState: unknown; currentState: unknown }
+    ): void {
+        const result = update(this.initialState, this.currentState);
+        this.initialState = result.initialState as ExerciseState;
+        this.currentState = result.currentState as ExerciseState;
+    }
+
```

Comment:
Where is this function used, why does it get an unknown update function, and why does it have to be in the exerciseWrapper (god class)?

Reply:
I just noticed that we don't actually need the whole "migrate in memory", so I've removed the respective code. For reference, still the answers to the questions:

1. Why does it have to be in `ExerciseWrapper`? We have to assign to `private` fields.
2. It is (was) used in an actual migration that was added in #477.
3. `unknown` because the idea is (was) to use it to migrate between state versions. Let's say we're migrating from version `7` to `8` when the current version is `9`. `9` added `someMysticalProperty` to the state. When we say these are `ExerciseState`s, TS would allow us to access `someMysticalProperty` although we don't have it as we're using an older state version at this point. I agree that this may not be the best solution. A similar case arises in #477, so let's discuss it further over there, if necessary. The only solution I'd see for this would be to include `declare` types for outdated state versions, which in itself is not great.